cli/cmd/plugin/ui/web/tanzu-ui/src/shared/services/awsDeployment.tsx

```diff
@@ -106,6 +100,24 @@ const useAwsDeployment = () => {
         navigateToProgress();
     };
 
+    const createAZPayLoadObject = () => {
+        const azs: SelectedAvailabiltyZoneData[] = [];
```

Comment:
Instead of defining the SelectedAvailabilityZoneData type you can use the AWSNodeAz type defined in the swagger spec. You'll find this in /src/swagger-api/models/AWSNodeAz.ts

Reply:
Okay, just looked a little closer. I see that you are using the generated nodeType object from RESOURCES when constructing the payload...to populate the controlPlaneNodeType.

I think maybe a slightly cleaner solution would be to modify this to initControlPlaneNodeType so that instead of storing a map in resources we can just do the calculation when the Node Profile value is changed, and then store the controlPlaneNodeType value in the FORM section.

Reply:
Yes sir. Used `AWSNodeAz`